### ISSUE TITLE
Arduino Pro Mini

### ISSUE THREAD

**Issue by themuck** [kriswiner/MPU9250#8]
Hallo,
i have an Arduino Pro Mini, MPU-9250 Board and Nokia 5110 Display, but the code works only without serial debug.

**Comment by kriswiner** [comment]
Yes, I have seen this problem with the Pro Mini. I think there is either a memory limitation or a timing conflict. I never really understood it. Sometimes the difference between working and not working was a single Serial.print! It's not a flaw in the sketch, but a flaw in the AVR platform. This is one of the many reasons I no longer use the Pro Mini and have moved almost exclusively to the Teensy 3.1.

**Comment by themuck** [comment]
OK thanks, what i want to do is to have all 3 axis in a stable oriantation form  -/+180°. But i am not realy shure how i must calculate it now. I think the "ypr" values shows this but this not realy working.

tempCount = readTempData();  // Read the adc values 
is inside  if(SerialDebug)... so the 5110 shows no tempreature.

i turn around X roll goes -180 to 0 to 180, i think it is OK
i turn around Z yaw goes -180 to 0 to 180, for this it was importent to use (max + min) / 2 to schift magnetic to 0.
i turn around Y pitch goes -90 to 0 to 90, i think this is ok

when i pitch ,yaw value is wrong and run to another Value 
when i roll , yaw value is wrong and run to anther Value.

I use a Mini Pro 5V with 16MHz 

here is my code https://github.com/themuck/MPU-9250-Arduino-Pro-Mini---Nokia-5110/blob/master/MPU%209250%20Arudino%20Pro%20Mini.ino

**Comment by themuck** [comment]
//MadgwickQuaternionUpdate(ax, ay, az, gx_PI/180.0f, gy_PI/180.0f, gz_PI/180.0f,  mx,  my, mz);
    MadgwickQuaternionUpdate(ax, ay, -az, gx_PI/180.0f, gy_PI/180.0f, gz_PI/180.0f,  my,  mx, mz); // change mx & my, and negate az to have all axes in one direction!!!

but now yaw swings over, and when i roll or pitch it change  :(

**Comment by themuck** [comment]
OK, now i checked all and use the https://github.com/richards-tech/RTIMULib-Arduino. 
Everything works...

**Comment by kriswiner** [comment]
I'm not sure what kind of problems you are having but I believe the Nokia 5110 is a 3V3-only display. I wouldn't use 5 V.

**Comment by themuck** [comment]
Now i think it works...
